File: sorting-snippets/bubblesort/bubblesort.cpp
#include <iostream>
#include <stdio.h>
#include <vector>

void bubblesort(std::vector<int>& a);
void printVector(std::vector<int> const &input);

int main() {
    std::vector<int> a {1,4,2,7,1,45,1,7,2,2};
    bubblesort(a);
    printVector(a);
    return 0;
}

void bubblesort(std::vector<int>& a) {
    for(unsigned int i = 0; i < a.size(); i++)
    {
        bool swapped = false;
        for(unsigned int j = 0; j < a.size() - i -1; j++)
        {
            if (a[j] > a[j+1]) {
                swapped = true;
                int tmp = a[j];
                a[j] = a[j+1];
                a[j+1] = tmp;
            }
        }
        if (!swapped) break;
    }
}

void printVector(std::vector<int> const &input)
{
	for (int i = 0; i < input.size(); i++) {
		std::cout << input.at(i) << ' ';
	}
}

File: sorting-snippets/bubblesort/bubblesort.cs
using System;

namespace BubbleSort
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] array = {2,2,5,2,6,1,1,6,78,4,8,3,2};
            Bubblesort(array);
            Console.WriteLine(string.Join(",", array));
        }

        static void Bubblesort(int[] array)
        {
            for (int i = 0; i < array.Length; i++)
            {
                bool swapped = false;
                for (int j = 0; j < array.Length-i-1; j++)
                {
                    if (array[j] > array[j+1])
                    {
                        swapped = true;
                        int tmp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = tmp;
                    }
                }

                if (!swapped) break;
            }
        }
    }
}
